_includes/master/manifests/calico-node.yaml

```diff
@@ -494,6 +496,9 @@ spec:
             type: DirectoryOrCreate
             path: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/nodeagent~uds
 {{- end }}
+        - name: bpffs
```

Comment:
What happens if BPF isn't enabled?  Will this folder exist?

Reply:
If BPF is not enabled in the kernel (`CONFIG_BPF_SYSCALL=n`) or if it is but the kernel is older than 4.4 (without support for BPFfs), the directory won't exist so I guess this would fail. Also, using `type: DirectoryOrCreate` is not possible because creating directories in `/sys/fs` is not allowed.

If it is enabled but BPFfs is not mounted we're fine because we'll mount it ourselves in Felix. One drawback is that the mount won't be propagated to the host so if Felix is restarted it will have to mount it and populate it again.

I think that to make sure we stay compatible with old kernels or kernels with BPF disabled we can't add this volume, which means Felix will repopulate it when we restart it, which is less efficient but shouldn't be a big deal.